content/blog/bare-metal-provisioning-on-hpe-greenlake-using-terraform.md

```diff
@@ -0,0 +1,359 @@
+2. Verify the installation with **terraform -help.** 
+
+   At this point, you are ready to start building your infrastructure description file. 
+
+# Deploy Compute Instance
+
+## Select Terraform provider with bare metal service configurations
+
+1. #### Export the following environment variables on your setup.
+
+   E﻿xport the Tenant ID:
+
+   ```shell
+   export HPEGL_TENANT_ID="<Tenant ID>
+   ```
+
+   Export the API Client credentials that you obtained when you created an API Client within HPE GreenLake Central:
+
+   ```shell
+   export HPEGL_USER_ID="<API Client ID>"
+   export HPEGL_USER_SECRET="<API Client Secret>"
+   export HPEGL_IAM_SERVICE_URL="<Issuer URL>"
+   ```
+
+   E﻿xport the Compute Group ID:
+
+   ```
+   # compute group/project ID
+   export HPEGL_METAL_PROJECT_ID="<Compute Group ID>"
+   ```
+
+    ﻿Export bare metal service REST URL:
+
+   ```
+   # Production Environment:  https://client.greenlake.hpe.com/api/metal
+   # Integration Environment: https://client.greenlake.hpe-gl-intg.com/api/metal
+   # local development: http://localhost:3002
+
+   export HPEGL_METAL_REST_URL="<Metal Service REST Base URL>"
+   ```
+2. #### Configure the Terraform provider.
+
+   Create an empty folder and put a file in it called main.tf with the following contents:
+
+   **main.tf:**
+
+   ```hcl
+   terraform {
+     required_providers {
+       hpegl = {
+         source  = "HPE/hpegl"
+         version = ">= 0.3.12"
+       }
+     }
+   }
+
+   provider "hpegl" {    
+     # metal block for configuring bare metal resources.
+     metal {   
+     } 
+   }
+   ```
+
+   T﻿his tells Terraform that you are going to be using HPE/hpegl as your provider and you are using bare metal service.
+
+## Write resource configuration for Compute Instance creation
+
+To deploy compute instance, you need to use the **hpegl_metal_host** terraform resource.
+
+***Note**:*
+
+* *compute instance is AKA host.* 
+* *compute instance type is AKA machine size.*
+
+The **hpegl_metal_host** resource supports many different arguments, but these are the required ones:
+
+* networks – List of networks. The list must always include any networks marked as '**required**' for that location. 
+* machine_size – Compute Instance type.
+* ssh – List of SSH keys to push to the host.
+* location – The data center location of where the compute instance will be provisioned.
+* Image – OS image in the form flavor@version.
+
+  Y﻿ou can also check the documentation [here](https://registry.terraform.io/providers/HPE/hpegl/latest/docs/resources/metal_host) to see all the required and optional fields.
+
+  Your next step with the TF file is to query the HPE GreenLake provider to collect the above-required information for creating a host. For this, you will use the Terraform data statements.  
+
+### Querying for available OS images
+
+In order to list the available OS images for OS, add the below data statements in your terraform file ***main.tf***:
```

Comment:
Terraform should have an initial capital letter as it's the name of the product here modifying the word "file"